framework/src/main/java/org/tron/core/services/filter/HttpInterceptor.java

```diff
@@ -0,0 +1,102 @@
+package org.tron.core.services.filter;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import lombok.extern.slf4j.Slf4j;
+import org.tron.core.metrics.MetricsKey;
+import org.tron.core.metrics.MetricsUtil;
+
+@Slf4j(topic = "httpIntercetpor")
+public class HttpInterceptor implements Filter {
+
+  private static final Map<String, Set<String>> EndpointMeterNameList = new HashMap<>();
+  private String endpoint;
+
+
+  public static Map<String, Set<String>> getEndpointList() {
+    return EndpointMeterNameList;
+  }
+
+
+  @Override public void init(FilterConfig filterConfig) throws ServletException {
+
+  }
+
+  @Override
+  public void doFilter(ServletRequest request, ServletResponse response, FilterChain chain)
+      throws IOException, ServletException {
+
+    try {
+      if (request instanceof HttpServletRequest) {
+        endpoint = ((HttpServletRequest) request).getRequestURI();
+        String endpointQPS = MetricsKey.NET_API_DETAIL_ENDPOINT_QPS + "." + endpoint;
```

Comment:
"."  can be added to the MetricsKey.NET_API_DETAIL_ENDPOINT_QPS

Reply:
good